pkg/i2gw/providers/ingressnginx/resource_reader.go

```diff
@@ -34,12 +41,51 @@ func newResourceReader(conf *i2gw.ProviderConf) *resourceReader {
 	}
 }
 
-func (r *resourceReader) ReadResourcesFromCluster(_ context.Context) error {
-	// ingress-nginx does not have any CRDs.
-	return nil
+func (r *resourceReader) readResourcesFromCluster(ctx context.Context) (*storage, error) {
```

Comment:
I'd suggest to move out `readResourcesFromCluster` logic to a `common` package with the filter option, because this logic is used by both `ingressnginx` & `kong` providers exactly.

Reply:
I thought about it, we can move this logic to common as `readIngressesFromCluster`. readResourcesFromCluster should still be a local function on the provider to orchestrate the read of whatever resources they need. 

wdyt?

Reply:
sgtm